BZ2029279: Removes the --index-filter-by-os flag (#48438)
openshift/openshift-docs: modules/olm-mirroring-catalog-colocated.adoc

    <mirror_registry>:<port>/<namespace> \ <2>
    [-a ${REG_CREDS}] \ <3>
    [--insecure] \ <4>

>     [--index-filter-by-os='<platform>/<arch>'] \ <5>

Reviewer: The `--index-filter-by-os='<platform>/<arch>'` still work, we should NOT remove it. But, the `--index-filter-by-os='.*'` doesn't work. We should remove the `.*` described in https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-installation-images.html#olm-mirror-catalog-colocated_installing-mirroring-installation-images, like the screenshot shows:
<img width="477" alt="image" src="https://user-images.githubusercontent.com/15416633/181669595-7b731852-ca8e-44ee-84c0-a54eb48c6728.png">

Author: Understood. Addressed. Thanks!

----
<1> Specify the index image for the catalog that you want to mirror. For example, this might be a pruned index image that you created previously, or one of the source index images for the default catalogs, such as `{index-image-pullspec}`.
<2> Specify the fully qualified domain name (FQDN) for the target registry and namespace to mirror the Operator contents to, where `<namespace>` is any existing namespace on the registry. For example, you might create an `olm-mirror` namespace to push all mirrored content to.
<3> Optional: If required, specify the location of your registry credentials file.
`{REG_CREDS}` is required for `registry.redhat.io`.
<4> Optional: If you do not want to configure trust for the target registry, add the `--insecure` flag.

> <5> Optional: Generate only the manifests required for mirroring, and do not actually mirror the image content to a registry. This option can be useful for reviewing what will be mirrored, and it allows you to make any changes to the mapping list if you require only a subset of packages. You can then use the `mapping.txt` file with the `oc image mirror` command to mirror the modified list of images in a later step. This flag is intended for only advanced selective mirroring of content from the catalog; the `opm index prune` command, if you used it previously to prune the index image, is suitable for most catalog management use cases.

Reviewer 2: ```suggestion
<5> Optional: Generate only the manifests required for mirroring without actually mirroring the image content to a registry. This option can be useful for reviewing what will be mirrored and lets you make any changes to the mapping list, if you require only a subset of packages. You can then use the `mapping.txt` file with the `oc image mirror` command to mirror the modified list of images. This flag is intended for only advanced selective mirroring of content from the catalog; the `opm index prune` command, if you used it previously to prune the index image, is suitable for most catalog management use cases.
```

Author: Addressed

<3> Optional: If required, specify the location of your registry credentials file.
`{REG_CREDS}` is required for `registry.redhat.io`.
<4> Optional: If you do not want to configure trust for the target registry, add the `--insecure` flag.
<5> Optional: Specify which platform and architecture of the index image to select when multiple variants are available. Images are passed as `'<platform>/<arch>[/<variant>]'`. This does not apply to images referenced by the index. Valid values are `linux/amd64`, `linux/ppc64le`, `linux/s390x`, `linux/arm64`.

> <6> Optional: Generate only the manifests required for mirroring without actually mirroring the image content to a registry. This option can be useful for reviewing what will be mirrored, and lets you to make any changes to the mapping list, if you require only a subset of packages. You can then use the `mapping.txt` file with the `oc image mirror` command to mirror the modified list of images in a later step. This flag is intended for only advanced selective mirroring of content from the catalog; the `opm index prune` command, if you used it previously to prune the index image, is suitable for most catalog management use cases.

Reviewer 3: Change: "and lets you to make any changes.." to "and lets you make any changes..." and allows you to make any changes..."

Reviewer 2: > Change: "and lets you to make any changes.." to "and lets you make any changes..." and allows you to make any changes..."

@lpettyjo This was part of an early copy edit. It should read `and lets you make any changes`. Good catch.

Author: @lpettyjo Addressed
